### PR TITLE
Enable `header_guard_check` (and `--fix`) in `ci/format`

### PR DESCRIPTION
**Pull request by matanlurey**
Closes https://github.com/flutter/flutter/issues/133415.

This includes a performance optimization for the runner itself, where `--include=...`s are used as a base to check, versus checking _every_ file and seeing if it exists in a list. This works better with what is expected in `ci/bin/format.dart` without any contract change.

Also added a few more functional tests of the system.

### CODE REVIEW

File: tools/header_guard_check/lib/header_guard_check.dart [hunk 1]
```diff
@@ -45,14 +48,20 @@ final class HeaderGuardCheck {
   /// Path directories to exclude from the check.
   final List<String> exclude;
 
+  /// Stdout.
+  final StringSink _stdOut;
+
+  /// Stderr.
+  final StringSink _stdErr;
```
Comment (cbracken) [inline tools/header_guard_check/lib/header_guard_check.dart:55]: pedantic nit: why not just `_stdout`, `_stderr` to match capitalisation elsewhere, and in the dart IO APIs?

Comment (matanlurey) [reply tools/header_guard_check/lib/header_guard_check.dart:55]: Because I hate myself I suppose.

**Review by cbracken — APPROVED** [review]
nice! Thanks for fixing up/enforcing this.

**Comment by matanlurey** [comment]
@gaaclarke I am going to land this to get enforcement in before header files are added/changed, but please still leave comments and I'll send a follow-up PR. Thanks!

**Pull request merged**